web/__tests__/pages/api/post/post.int.test.ts

```diff
@@ -181,112 +183,101 @@ describe("[API] Post - Integration Test", () => {
-        _id: id,
-        attachments,
+        ...post,
+        _id: uncoveredId,
+        covered: false,
       });
       expect(createUncoveredPost).not.toBeNull();
-      let goodWithArray = [];
-      for (const goodWithKey in goodWithMap) {
-        if (createCoveredPost[goodWithKey] == "yes") {
-          goodWithArray.push(goodWithMap[goodWithKey]);
-        }
-      }
-      const postFilterSchema = {
-        type: [coveredPost.type],
-        breed: coveredPost.breed,
-        age: [coveredPost.age],
-        size: [coveredPost.size],
-        gender: [coveredPost.gender],
-        behavioral: coveredPost.behavioral,
-        goodWith: goodWithArray,
+      const postFilters = {
+        type: Object.values(FosterType),
+        breed: Object.values(Breed),
+        age: Object.values(Age),
+        size: Object.values(Size),
+        gender: Object.values(Gender),
+        behavioral: Object.values(Behavioral),
+        goodWith: [],
       };
       const filteredPosts = await caller.post.getFilteredPosts({
-        postFilterSchema: postFilterSchema,
+        postFilterSchema: postFilters,
       });
       expect(filteredPosts).not.toBeNull();
       expect(filteredPosts.length).toBe(2);
       const coveredPosts = await caller.post.getFilteredPosts({
-        postFilterSchema: postFilterSchema,
+        postFilterSchema: postFilters,
         covered: true,
       });
       expect(coveredPosts).not.toBeNull();
       expect(coveredPosts.length).toBe(1);
+      expect(coveredPosts[0]._id).toMatchObject(coveredId);
       const uncoveredPosts = await caller.post.getFilteredPosts({
-        postFilterSchema: postFilterSchema,
+        postFilterSchema: postFilters,
         covered: false,
       });
       expect(uncoveredPosts).not.toBeNull();
       expect(uncoveredPosts.length).toBe(1);
+      expect(uncoveredPosts[0]._id).toMatchObject(uncoveredId);
     });
 
     test("random covered-uncovered", async () => {
-      let randomPost = createRandomPost();
-      const numPosts = Math.ceil(Math.random() * 10);
-      let numCovered = 0;
-      let numUncovered = 0;
+      const randPosts = randomPosts;
+      let allIds = [];
+      let covered = [];
+      let uncovered = [];
       let createPost;
-      for (let i = 0; i < numPosts; i++) {
-        const curr = Math.floor(Math.random() * 2);
-        if (curr == 0) {
-          randomPost.covered = false;
-          numUncovered++;
-        } else {
-          randomPost.covered = true;
-          numCovered++;
-        }
+      for (let i = 0; i < 10; i++) {
         const id = new mongoose.Types.ObjectId();
+        allIds.push(id.toString());
+        randPosts[i].covered
+          ? covered.push(id.toString())
+          : uncovered.push(id.toString());
         createPost = await Post.create({
-          ...randomPost,
+          ...randPosts[i],
           _id: id,
-          attachments,
         });
         expect(createPost).not.toBeNull();
       }
```

Comment:
Use `Post.insertMany` instead of looping 10 times and individually adding the documents